scripts/export_large_tsv/export_large_tsv_python3.py

```diff
@@ -0,0 +1,110 @@
+    """Download large TSV file from Terra workspace by designated number of rows."""
+    # get all entity types in workspace using API call
+    # API = https://api.firecloud.org/#!/Entities/getEntityTypes
+    request = fapi.list_entity_types(project, workspace)
+    if request.status_code != 200:
+        print(request.text)
+
+    # get/report # of entities + associated attributes(column names) of input entity type
+    entity_types_json = request.json()
+    entity_count = entity_types_json[entity]["count"]
+    attribute_names = entity_types_json[entity]["attributeNames"]
+    entity_id = entity_types_json[entity]["idName"]
+    # add the entity_id value to list of attributes (not a default attribute of API response)
+    attribute_names.insert(0, entity_id)
+
+    print(f'{entity_count} {entity}(s) to export.')
+
+    with open(tsv_name, "w") as tsvout:
+        # add header with attribute values to tsv
+        tsvout.write("\t".join(attribute_names) + "\n")
+        # set starting row value and calculate number of pages
+        row_num = 0
+        num_pages = int(math.ceil(float(entity_count) / page_size))
+
+        # get entities by page where each page has page_size # of rows using API call
+        # API = https://api.firecloud.org/#!/Entities/entityQuery
+        print(f'Getting all {num_pages} pages of entity data.')
+        all_page_responses = []
+        for page in tqdm(range(1, num_pages + 1)):
+            all_page_responses.append(get_entity_by_page(project, workspace, entity, page, page_size))
+
+        # for each response(page) in all_page_responses[] - contains parameter metadata
+        print(f'Writing {entity_count} attributes to tsv file.')
+        for page_response in tqdm(all_page_responses):
+            # for each set of attributes in results (no parameters) get attribute names and entity_id(name)
+            for entity_json in page_response["results"]:
+                attributes = entity_json["attributes"]
+                name = entity_json["name"]
+                # add name and value to dictionary of attributes
+                attributes[entity_id] = name
+
+                values = []
+                # for each attribute(column name) in list of attribute names(all columns for entity)
+                for attribute_name in attribute_names:
+                    value = ""
+                    # if entity's attribute(column) is in list of attributes from response, set response's attribute value
+                    if attribute_name in attributes:
+                        value = attributes[attribute_name]
+
+                    values.append(value)
+
+                tsvout.write("\t".join(values) + "\n")
+                row_num += 1
+    print(f'Finished exporting {entity}(s) to tsv with name {tsv_name}.')
+
+
+def get_entity_by_page(project, workspace, entity, page, page_size, sort_direction='asc', filter_terms=None):
```

Comment:
I think it's best practice to have supporting functions appear at the top, and have the "main" function be the last one. (I may be wrong, but that's how I've seen it!) Could you move this up to the top of the script?